languages/rust/oso/Cargo.toml

```diff
@@ -41,14 +41,18 @@ rustyline = { version = "6.3.0", optional = true }
 rustyline-derive = { version = "0.3.1", optional = true }
 lazy_static = "1.4.0"
 
+uuid = { version = "0.6.5", optional = true }
+
 [dev-dependencies]
 anyhow = "1.0.33"
 criterion = "0.3.3"
 oso-derive = { path = "../oso-derive", version = "=0.8.2" }
 tempfile = "3.1.0"
 static_assertions = "1.1.0"
+uuid = "0.6.5"
```

Comment:
> Additionally, is there a way to specify that this dev dependency should only be built when the uuid feature is enabled?

Actually, I think this is already the case. The test feature-gates it so it shouldn't be necessary here.

Reply:
> Forgive my ignorance, but will someone who's using oso with a different major version of uuid (such as the current 0.8.1) be able to use this feature?

No. @johnalbert - we discussed this on a call, can we make the feature `uuid_v06` or similar (are there precedents in other crates?) to make this clear?

Reply:
Here's an example - https://github.com/sfackler/rust-postgres/blob/master/postgres-types/Cargo.toml#L37

Reply:
@gj you _can_ use newer versions of `uuid`, and I have in the project where I'm integrating Oso.  Rust sees each version as it's own type, so they won't share the `PolarClass` implementation.  Only the type provided by `0.6.5` will have it.

In my case, I get two versions from my own crate's dependencies - Diesel ORM (`0.6.5`) and Juniper GraphQL framework (`0.7.4`).  Diesel won't support anything beyond `0.6.5` until a new major version is released - https://github.com/diesel-rs/diesel/issues/1900.

Where the two types meet, the value must be converted from one to the other.  Not ideal.  I minimize this by performing conversion as late as possible (when storing value provided as input to gql in db, or when providing value from db in response to gql query).  And in the case of Oso, don't perform any conversion, since I use values directly from the database.

This was my reasoning for using `0.6.5`.  Diesel is, I think, the most popular ORM for Rust (can only think of one other, and I never see it mentioned).  So, major projects that are using the database will likely also be using Diesel.  I could definitely see it making sense to support later versions as well, but just wanted to mention why I used it here so you have some context.

Reply:
Okay on above comments.  Makes sense.  Only thing I would ask is regarding below comments about making these part of a separate crate - if that's the route you think is best, let me know.  Makes the most sense that I would refactor that way while making these other changes.

Reply:
I think it's fine for now. Agree to putting in a separate module at least. And agree with @gj that there's no need to group them for now.

In the longer term, we have some ideas for avoiding the need for this at all. But would prefer to get this merged in first so we can at least cover this in the short term.

Reply:
Yep, second @samscott89. Let's get this change in for now behind the `uuid-06` flag (or similar). Thanks for the contribution!